text/0059-geo-search.md

```diff
@@ -0,0 +1,232 @@
+
+### III. Technical Explanations
+
+#### **As a developer, I want to add geo-spatial coordinates to a document so that the document can be geo-searchable.**
+
+- Introduce a reserved field `_geo` for documents to store geo spatial data from an **object** made of `lat` and `lng` fields for a **JSON format**.
+- Introduce a reserved column `_geo` for documents to store geo spatial data from a **string** made of `lat,lng` for a **CSV format**.
+
+##### **JSON Format**
+
+**`_geo` field definition**
+
+- Name: `_geo`
+- Type: Object
+- Format: `{lat:float, lng:float}`
+- Not required
+
+> 💡 if `_geo` is found in the document payload, `lat` and `lng` are required.
+> 💡 `lat` and `lng` must be of float value.
+
+##### **CSV Format**
+
+Following the format already defined in the https://github.com/meilisearch/specifications/pull/28/files specification for document indexing from a CSV format. A reserved column `_geo` can be added to specify the geographical coordinates of a document.
+
+csv format example
+```
+"id:number","label","brand","_geo"
+"1","F40","Ferrari","48.862725,2.287592"
+```
+
+**`_geo` column definition**
+
+- Name: `_geo`
+- Type: String
+- Format: `"lat:float,lng:float"`
+- Not required
+
+#### POST Add or replace documents `/indexes/{indexUid}/documents`
+
+##### Request body
+```
+[
+    {
+        "id": 1,
+        "label": "F40",
+        "brand": "Ferrari",
+        "_geo": {
+            "lat": 48.862725,
+            "lng": 2.287592
+        }
+    }
+]
+```
+
+##### 202 Accepted - Response body
+
+```
+{
+    "updateId": 1
+}
+```
+
+#### PUT Add or replace documents `/indexes/{indexUid}/documents`
+
+##### Request body
+```
+[
+    {
+        "id": 1,
+        "brand": "F40 LM",
+        "brand": "Ferrari",
+        "_geo": {
+            "lat": 48.862725,
+            "lng": 2.287592
+        }
+    }
+]
+```
+
+##### 202 Accepted - Response body
+
+```
+{
+    "updateId": 2
+}
+```
+
+> 🔴 Giving a bad formed `_geo` that do not conform to the format causes the `update` payload to fail. A new `invalid_geo_field` error is given in the `update` object.
+
+##### `invalid_geo_field` error
+
+| field     | value                                                                                                                |
+|-----------|----------------------------------------------------------------------------------------------------------------------|
+| message   | The _geo field is invalid.                                                                                           |
+| errorCode | invalid_geo_field                                                                                                    |
+| errorType | invalid_request_error                                                                                                |
+| errorLink | *Link to the dedicated error page. This page should give further links to indicate the expected JSON and CSV format* |
```

Comment:
These will eventually be changed to follow spec #61, correct?

Reply:
If the error format spec is accepted and approved first, I would take the time to update it for in-progress/ready-for-review PR specifications however spec files could also be modified on the release PR that will prepare and summarize all implemented specs.

But to answer you yes it would be changed. 

I wouldn't update all the old specs that are already implemented. The specifications are epic mainly used by us to define what should be done in a specific iteration in an open way.